### PR TITLE
Cw 335 fixes for limit zero ver2

### PR DESCRIPTION
**Pull request by aleksander-bulanowski**

### CODE REVIEW

**Comment by michalzurawski** [comment]
it's worth mention one drawback of this solution pointed out by @aleksander-bulanowski - a conversion between int and uint. It's possible to pass a negative value which then would be translated to uint

**Comment by morrisson** [comment]
Please squash commits and rebase.

File: db/file/file.go [hunk 1]
```diff
@@ -300,7 +301,7 @@ func (tx *Transaction) List(s *schema.Schema, filter transaction.Filter, options
 
 		if pg != nil {
 			sort.Sort(byPaginator{list, pg})
-			if pg.Limit > 0 {
+			if pg.Limit != math.MaxUint64 {
 				list = list[:pg.Limit]
```
Comment (jkmar) [inline db/file/file.go]: this will panic if `pg.Limit > len(list)`

File: extension/goext/schemas.go [hunk 2]
```diff
@@ -53,7 +53,7 @@ type Filter map[string]interface{}
 type Paginator struct {
 	Key    string
 	Order  string
-	Limit  uint64
+	Limit  MaybeInt
```
Comment (jkmar) [inline extension/goext/schemas.go]: Since was uint64 before I'd rather made a new type for it: `MaybeUint64`.

**Comment by morrisson** [comment]
@aleksander-bulanowski you are working on #643 so we can close this PR without the merge, right?

**Comment by aleksander-bulanowski** [comment]
Since https://github.com/cloudwan/gohan/pull/643 was merged there is no need to merge this PR.

**Pull request closed without merging**